spec/features/changing_answers_spec.rb

```diff
@@ -1,284 +1,230 @@
+    find("a[href='#{claim_path(MathsAndPhysics.routing_name, "supply-teacher")}']").click
 
-      choose "Yes"
-      click_on "Continue"
-    end
+    expect(find("#claim_eligibility_attributes_employed_as_supply_teacher_true").checked?).to eq(true)
 
-    scenario "and spent less than half their time performing leadership duties" do
-      expect(eligibility.reload.had_leadership_position).to eq(true)
-      expect(eligibility.reload.mostly_performed_leadership_duties).to eq(nil)
+    click_on "Continue"
 
-      expect(current_path).to eq(claim_path(StudentLoans.routing_name, "mostly-performed-leadership-duties"))
+    expect(claim.eligibility.reload.employed_as_supply_teacher).to eq(true)
+    expect(claim.eligibility.has_entire_term_contract).to eq(true)
+    expect(claim.eligibility.employed_directly).to eq(true)
 
-      choose "No"
-      click_on "Continue"
+    expect(current_path).to eq(claim_path(MathsAndPhysics.routing_name, "check-your-answers"))
+  end
 
-      expect(eligibility.reload.mostly_performed_leadership_duties).to eq(false)
-      expect(current_path).to eq(claim_path(StudentLoans.routing_name, "check-your-answers"))
-    end
+  scenario "when changing the student loan repayment amount the user can change answer and it preserves two decimal places" do
+    claim = start_student_loans_claim
+    claim.update!(attributes_for(:claim, :submittable))
+    claim.eligibility.update!(attributes_for(:student_loans_eligibility, :eligible, student_loan_repayment_amount: 100.1))
+    visit claim_path(StudentLoans.routing_name, "check-your-answers")
 
-    scenario "and spent more than half their time performing leadership duties" do
-      choose "Yes"
-      click_on "Continue"
+    expect(page).to have_content("£100.10")
+    find("a[href='#{claim_path(StudentLoans.routing_name, "student-loan-amount")}']").click
 
-      expect(eligibility.reload.mostly_performed_leadership_duties).to eq(true)
+    expect(find("#claim_eligibility_attributes_student_loan_repayment_amount").value).to eq("100.10")
+    fill_in I18n.t("student_loans.questions.student_loan_amount"), with: "150.20"
+    click_on "Continue"
 
-      expect(page).to have_text("You’re not eligible")
-      expect(page).to have_text("You can only get this payment if you spent less than half your working hours performing leadership duties between 6 April 2018 and 5 April 2019.")
-    end
+    expect(page).to have_content("£150.20")
   end
 
-  context "Teacher changes claim school to another eligible school" do
-    let!(:new_claim_school) { create(:school, :student_loan_eligible, name: "Claim School") }
+  context "User changes fields that aren't related to eligibility" do
+    let(:claim) { Claim.order(:created_at).last }
+    let(:eligibility) { claim.eligibility }
 
     before do
-      find("a[href='#{claim_path(StudentLoans.routing_name, "claim-school")}']").click
-
-      choose_school new_claim_school
+      claim = start_student_loans_claim
```

Comment:
I feel like `start_student_loans_claim` could be inside the let as it is the same thing and this just seems to re-assign it here?

Reply:
You're right. I've had to change it to a `let!` so that `start_student_loans_claim` happens before the `it` starts.